kbase-extension/static/kbase/js/widgets/appWidgets2/input/newObjectInput.js

```diff
@@ -16,65 +16,28 @@ define([
         input = t('input');
 
     function factory(config) {
-        const options = {},
-            spec = config.parameterSpec,
+        const spec = config.parameterSpec,
             workspaceId = config.workspaceId,
             bus = config.bus,
             model = {
                 value: undefined,
             },
             runtime = Runtime.make(),
             otherOutputParams = config.closeParameters
-                ? config.closeParameters.filter((value) => {
-                      return value !== spec.id;
-                  })
+                ? config.closeParameters.filter((value) => value !== spec.id)
                 : [];
-        let parent, container, ui;
-
-        // Validate configuration.
-        // Nothing to do...
-
-        options.enabled = true;
-
-        /*
-         * If the parameter is optional, and is empty, return null.
-         * If it allows multiple values, wrap single results in an array
-         * There is a weird twist where if it ...
-         * well, hmm, the only consumer of this, isValid, expects the values
-         * to mirror the input rows, so we shouldn't really filter out any
-         * values.
-         */
+        let container, ui, autoChangeTimer;
 
         function getInputValue() {
             return ui.getElement('input-container.input').value;
         }
 
         function setModelValue(value) {
-            return Promise.try(() => {
```

Comment:
1. That doesn't need to be a Promise.
2. Return values do nothing.
3. Shouldn't need to rebuild the whole DOM after changing the value.